### ISSUE TITLE
sLaunch MOD 1.11 graphical corruption

### ISSUE THREAD

**Issue by kar0sh1** [aldostools/webMAN-MOD#445]
PS3: CECH-2001A / CECH-2101A
CFW: 4.86.1 Rebug Lite
WebMan MOD v1.47.31 (full version)

The very top of the sLaunch MOD 1.11 screen background shows tiny illegible white text on a black background which I believe is the name of each game in the scanned collection.  It's very noticeable with a large collection, but even a collection consisting of just a single game will cause this to happen on my systems (see images below).  This problem does not exist when using the sLaunch included with sMAN v1.12n.

I've done the following:
* used multiple PS3s
* tried the different variants of WebMan MOD (full/std)
* tried different artwork image formats/sizes
* tried with NO artwork
* nuked the /dev_hdd0/tmp dir
* loaded game ISOs (all IRD/redump valid) via hdd0 / usb ntfs / ps3netsrv
* tried all image options in setup.ps3's 'disable multiMAN covers' options list (MM COVERS, ICON0.PNG, DISC ICONS, ONLINE COVERS)
* removed spaces and special chars from all ISO filenames.
* replaced the HDD with a blank one (no partitions), partitioned/formatted via PS3, reinstalled the OS, installed webman MOD, added a single game ISO with no artwork to /dev_hdd0/PS3ISO.

It makes no difference what I try.  I always have that corruption in the background at the top of the screen.  This has been a problem on all my systems for every version of WebMan MOD I've tried (since about a year or so ago using rebug 4.84.1 REX).

Large collection example:
![large-set](https://user-images.githubusercontent.com/74883082/99922131-a0171e00-2ce3-11eb-8fc7-825185399577.jpg)

Single-game collection with no artwork (fresh install):
![single-game-zoomed](https://user-images.githubusercontent.com/74883082/99922198-1451c180-2ce4-11eb-8939-66da0d6c2838.jpg)

Please let me know what I can do to further test, etc.

Thanks for your help and all your work on this project.

**Comment by aldostools** [comment]
It's a weird graphical bug... It took me several hours to "fix" it.

You can try extracting this [slaunch.zip](https://github.com/aldostools/webMAN-MOD/files/5581317/slaunch.zip), then replace **slaunch.sprx** in /dev_hdd0/tmp/wm_res

or re-download the installer from the release page
https://github.com/aldostools/webMAN-MOD/releases

**Comment by kar0sh1** [comment]
Thanks Aldo.  I tested the new sprx on two systems and the problem still exists.  The only difference this time is that there is now a clean black area (about 5 lines or so) at the very top for a game list consisting of 1 to 117 games.  118 will create a small blank area on a new line below the clean black area.  119+ games will then add the small white text for each additional game in the collection as seen before.

On my primary system I just replaced the sprx and rebooted.  

On a second system I did the following:
* "Restore PS3" > Quick Format > reboot
* installed WebMan MOD 1.47.31 >> Full Version
* changed two settings via setup.ps3: 
** adjusted fan profile (checked 'lowest', set to 50%. PS2 Emu: 50%)
** disabled content scan at startup
* replaced original slaunch.sprx (MD5: ff4899b55f5ab4bf757f7d85e23a32e7) with provided sprx (MD5: 5dac2dccee1b9080bc256d8a358f9d91)
* hard reboot
* imported games >> "Refresh My Games & XMB"
* changed two more settings via setup.ps3 since artwork wasn't downloading over ps3netsrv:
** unchecked "PARAM.SFO"
** checked "Include the ID as part of the title of the game"
* changed another setup.ps3 option to help with cover art for large collection:
** IDPS & MEM Setup >> Plugin memory usage [MC]: 4-bg 

The results were the same on both systems.  Here are screen pics from the second test system.  Sorry about the quality, but you should be able to see what's going on in the top left corner of the full images.

117-game collection with artwork.  clean black area at the top (hard to see in picture, but there is about 5mm of black space):
![117-actual-w-artwork](https://user-images.githubusercontent.com/74883082/100013837-57ff0680-2d8a-11eb-823e-6d60f3c94ff7.jpg)

119-game collection with artwork.  single white text entry below clean area:
![119-actual-w-artwork](https://user-images.githubusercontent.com/74883082/100017353-9cd96c00-2d8f-11eb-93ff-26d86fbf1ab0.jpg)

large set with artwork:
![large-set-w-artwork](https://user-images.githubusercontent.com/74883082/100017422-b7abe080-2d8f-11eb-9ccc-bec9e917da85.jpg)

**Comment by aldostools** [comment]
Thank you for your feedback.

Actually the "fix" was to replace the first 8 lines of the screen with a solid black line.
On my system the garbage produced random white dots in these 8 lines, so the workaround was to make that area black.

The image corruption is an issue that I don't understand what produce it.
I have invested over 14 hours (7 only yesterday) trying to fix the issue, and that was the best result that I could achieve.

The issue seems to happen when the background image is flipped to the next video frame: each pixel must be copied individually. But the memory area seems to be overwritten producing that garbage.

For the moment I don't plan to invest more time in this issue. I will leave this issue to any dev in the community that want to take this challenge.

**Comment by kar0sh1** [comment]
Bummer.  Thanks for looking at this all the same.  This bug doesn't affect functionality in any way that I can see.  It's just a bit of an eyesore.  I was hoping it would be as easy as taking Dean's latest sMAN sLaunch code and merging it with your own.  I personally prefer your implementation of sLaunch (favorites, better navigation, bypass of platforms with no games, sorting by filename, nicer backgrounds, temperature display), but I'd be willing to trade a few of these extras (not favorites though) for a clean look.

Here's to hoping someone can help out.  Thanks again for all your great work.

**Comment by aldostools** [comment]
After discuss this subject with you, I had a new idea for a proper fix which seems to be working (at least it did in my console).

You can re-download this installer and install it again:
https://github.com/aldostools/webMAN-MOD/releases/download/1.47.31/webMAN_MOD_1.47.31_Installer.pkg

**Comment by kar0sh1** [comment]
Thanks for the new build.  I just ran the same tests as before on this one.  There is certainly an improvement somewhat over the previous iteration, as there appears to be zero issue with my collection up to 125 games.  No corruption, no black bar, looks great.  Any more than 125 games, however, and I see the glitchy garbage.  My favorites list right now is smaller than 125 games, so when in that view it looks perfect.  If this workaround could be adjusted to deal with collections in the thousands, it may do the trick.

splitscreen of two PS3s w/ same game list.. old version (left) new version (right):
![before-after](https://user-images.githubusercontent.com/74883082/100191142-8113ba00-2ea4-11eb-9f9e-9d996b420102.jpg)

125-game collection.  looks perfect!  any more and the corruption appears.
![125-limit](https://user-images.githubusercontent.com/74883082/100191173-8ec93f80-2ea4-11eb-9ee2-b966c77f0c93.jpg)

**Comment by aldostools** [comment]
Thank you for your feedback. I think I finally found the issue. You can try this build
[slaunch.zip](https://github.com/aldostools/webMAN-MOD/files/5597143/slaunch.zip)

**Comment by kar0sh1** [comment]
I think you've fixed it!  Same tests as before.  I replaced the older slaunch.sprx with this new one (MD5: cf478ae4050562b858aee2b7d48bdf07) and hard rebooted.  Everything looks great.  No corruption, no black area, 1300+ game list, same load speed as before (identical to sMAN).

Splitscreen of two PS3s: before / after same 1071-game collection:
![before-after2b](https://user-images.githubusercontent.com/74883082/100270846-848f5b80-2f0d-11eb-98ff-8d81df0c4cab.jpg)

I wasn't able to test with a larger collection (PSX: 1500+ games) because the "Refresh My Games & XMB" scan would either hang indefinitely, freeze the system, not scan all games, or mess up the XMB XML (mygames.xml?).  I tried all permutations of memory profiles with the same results.  This is a totally unrelated problem, and very large collections are a bit of a nightmare to navigate anyway without a search filter or much faster page loading.  For most reasonably-sized collections, though, everything works and looks great now.  Thanks a ton, Aldo!  Awesome job.  I'll keep any eye out for your next release build.
